### PR TITLE
*: add support of custom timeouts

### LINKED ISSUE #189: Support custom timeout when evicting leaders
We currently hard code the timeout settings in `pdapi`, but for some large clusters, the leader transferring process may take far more long time (and that's acceptable for the circumstances).

We need to add support of custom timeout values for those operations. Rolling upgrade is in my mind, but there might be more places.

### PR DESCRIPTION
**Pull request by AstroProfundis**
- Add `--transfer-timeout` argument to set custom timeout when transferring leaders (Close #189 )
 - Add `--ssh-timeout` argument to set custom timeout when connecting to SSH servers

### CODE REVIEW

File: pkg/operation/upgrade.go [hunk 1]
```diff
@@ -39,6 +38,11 @@ func Upgrade(
 
 	leaderAware := set.NewStringSet(meta.ComponentPD, meta.ComponentTiKV)
 
+	timeoutOpt := &utils.RetryOption{
+		Timeout: time.Second * time.Duration(options.Timeout),
+		Delay:   time.Second * 5,
```
Comment (july2993) [inline pkg/operation/upgrade.go:43]: 1 second?

Comment (AstroProfundis) [reply pkg/operation/upgrade.go:43]: Would 1s delay be too frequent?

File: pkg/ansible/dirs.go [hunk 2]
```diff
@@ -29,14 +30,15 @@ var (
...
-func parseDirs(host *aini.Host, ins meta.InstanceSpec) (meta.InstanceSpec, error) {
+func parseDirs(host *aini.Host, ins meta.InstanceSpec, sshTimeout int64) (meta.InstanceSpec, error) {
 	hostName, sshPort := ins.SSH()
 
 	e := executor.NewSSHExecutor(executor.SSHConfig{
 		Host:    hostName,
 		Port:    sshPort,
 		User:    host.Vars["ansible_user"],
 		KeyFile: SSHKeyPath(), // ansible generated keyfile
+		Timeout: time.Second * time.Duration(sshTimeout),
```
Comment (lonng) [inline pkg/ansible/dirs.go:41]: Do we need to forbid to set the `sshTimeout` to zero?

File: pkg/operation/scale_in.go [hunk 3]
```diff
@@ -121,6 +122,11 @@ func ScaleIn(
 		return err
 	}
 
+	timeoutOpt := &utils.RetryOption{
+		Timeout: time.Second * time.Duration(options.Timeout),
```
Comment (lonng) [inline pkg/operation/scale_in.go:126]: ditto

File: pkg/ansible/dirs.go (hunk 2 above)
Comment (AstroProfundis) [reply pkg/ansible/dirs.go:41]: For `SSHConfig`, if the `Timeout` is zero, it is set to 5s by default (in `pkg/executor/ssh.go:87`).

File: pkg/operation/scale_in.go (hunk 3 above)
Comment (AstroProfundis) [reply pkg/operation/scale_in.go:126]: For `RetryOption`, if the `Timeout` is zero, `Retry()` will return an error. Currently we set timeouts before calling `Retry()` by initializing `RetryOption` objects. There is a default timeout of 10s but it is only used when calling `Retry()` without any `RetryOption` object passed.

Comment (AstroProfundis) [reply pkg/operation/scale_in.go:126]: I feel it's not reasonable to init a `RetryOption` object with `Delay` set but not setting `Timeout`.

**Pull request merged**